OpenRA.Game/Graphics/TerrainRenderer.cs

```diff
@@ -38,12 +47,24 @@ public TerrainRenderer(World world, WorldRenderer wr)
 
 		public void UpdateCell(CPos cell)
 		{
-			terrain.Update(cell, theater.TileSprite(mapTiles[cell]));
+			var tile = mapTiles[cell];
+			var palette = world.TileSet.Templates[tile.Type].Palette ?? TerrainPalette;
+			var sprite = theater.TileSprite(tile);
+			foreach (var kv in spriteLayers)
+			{
+				if (palette != kv.Key)
+					continue;
+
+				kv.Value.Update(cell, sprite);
+				break;
+			}
```

Comment:
Maybe this is copied code, but why is this a loop? It appears to be doing a `TryGetValue` on `spriteLayers`?